programs/mango-v4/src/instructions/serum3_edit_market.rs

```diff
@@ -0,0 +1,25 @@
+use anchor_lang::prelude::*;
+
+use crate::state::*;
+
+#[derive(Accounts)]
+#[instruction(market_index: Serum3MarketIndex)]
+pub struct Serum3EditMarket<'info> {
+    #[account(
+        has_one = admin,
+        constraint = group.load()?.serum3_supported()
+    )]
+    pub group: AccountLoader<'info, Group>,
+    pub admin: Signer<'info>,
+
+    #[account(
+        mut,
+        has_one = group
+    )]
+    pub market: AccountLoader<'info, Serum3Market>,
+}
+
+pub fn serum3_edit_market(ctx: Context<Serum3EditMarket>, reduce_only: bool) -> Result<()> {
+    ctx.accounts.market.load_mut()?.reduce_only = u8::from(reduce_only);
```

Comment:
What does it mean for a serum market to be reduce-only?

It makes sense to me for perp-markets to be reduce-only: users must reduce their perp base position. Banks can also be reduce-only: users can't increase borrows or increase deposits.

But spot markets have no state that could be reduced? I'd say we need a "can't place new orders on this market" flag, but that should be all?

Reply:
You have a good point, but how does one close deposits/borrows on token positions? If I remove my current serum market reduce only mode, alternative to closing token positions would only be to deposit/withdraw

Reply:
as discussed serum reduce only is very simple, doesnt support placing new orders